harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,840 @@
+                f"Still got {code} with {data}"
+            )
+
+        # Ping management ip address
+        mgmt_ip = ip_addresses[0]
+
+        ping_command = "ping -c 50 {0}".format(mgmt_ip)
+
+        _stdout, _stderr = self.ssh_client(
+            client, client_ip, node_user, node_password, ping_command, wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("64 bytes from {0}".format(mgmt_ip)) > 0, (
+            'Failed to ping VM management IP %s' % (mgmt_ip))
+
+        # ssh to host and execute command
+        _stdout, _stderr = self.ssh_jumpstart(
+            client, mgmt_ip, client_ip, node_user, node_password,
+            vm_credential["user"], vm_credential["password"], "ls")
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            'Failed to ssh to VM management IP %s' % (mgmt_ip))
+
+        # cleanup vm
+        delete_vm(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="delete_vlan_connection",
+                            depends=["mgmt_network_connection"])
+    def test_delete_vlan_from_multiple(self, api_client, request, client, unique_name,
+                                       vlan_network):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/delete-vlan-network-form/
+
+        Steps:
+        1. Create an external VLAN network
+        2. Create a new VM
+        3. Make sure that the network is set to the management network with masquerade as the type
+        4. Add another external VLAN management
+        5. Wait until the VM boot in running state
+        6. Delete the external VLAN from VM
+        7. Validate interface was removed with ip link status
+        8. Check can ping the VM on the management network
+        """
+
+        wait_timeout = request.config.getoption('--wait-timeout')
+        client_ip = request.config.getoption('--endpoint').strip('https://')
+
+        spec = api_client.vms.Spec(1, 2)
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+        unique_name = unique_name + "-delete-vlan"
+        # Create VM
+        spec.add_image(image_name, "default/" + image_name)
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # get data from running VM and transfer to spec
+        code, data = api_client.vms.get(unique_name)
+        spec = spec.from_dict(data)
+
+        # Switch to vlan network
+        spec.add_network("nic-1", "default/" + vlan_network['id'])
```

Comment:
I thought steps before here in not align to test plan, it described:
```
On a VM with both an external VLAN and a management VLAN delete the external VLAN via the web form
```
but not tell us to add network after the VM started.
I would suggest to add 2 NICs before create the VM.

And the comment looks weird, we are adding new NIC, not switch to.